cpp/include/fixed_point/fixed_point.hpp

```diff
@@ -0,0 +1,489 @@
+ * @brief Helper struct for constructing `fixed_point` when value is already
+ * shifted
+ *
+ * example: using decimal32 = fixed_point<int32_t, Radix::BASE_10>;
+ *          auto n = decimal32{scaled_integer{1001, 3}}; // n = 1.001
+ *
+ * @tparam Rep The representation type (either `int32_t` or `int64_t`)
+ *---------------------------------------------------------------------------**/
+template <typename Rep,
+          typename std::enable_if_t<is_supported_representation_type<Rep>()>* = nullptr>
+struct scaled_integer{
+    Rep value;
+    scale_type scale;
+    explicit scaled_integer(Rep v, scale_type s) : value(v), scale(s) {}
+};
+
+// Rep = representation type
+
+/**---------------------------------------------------------------------------*
+ * @brief A type for representing a number with a fixed amount of precision
+ *
+ * Currently, only binary and decimal `fixed_point` numbers are supported.
+ * Binary operations can only be performed with other `fixed_point` numbers
+ *
+ * @tparam Rep The representation type (either `int32_t` or `int64_t`)
+ * @tparam Rad The radix/base (either `Radix::BASE_2` or `Radix::BASE_10`)
+ *---------------------------------------------------------------------------**/
+template <typename Rep, Radix Rad>
+class fixed_point {
+
+    Rep        _value;
+    scale_type _scale;
+
+public:
+
+    /**---------------------------------------------------------------------------*
+    * @brief Constructor that will perform shifting to store value appropriately
+    *
+    * @tparam T The type that you are constructing from (integral or floating)
+    * @param value The value that will be constructed from
+    * @param scale The exponent that is applied to Rad to perform shifting
+    *---------------------------------------------------------------------------**/
+    template <typename T,
+              typename std::enable_if_t<is_supported_construction_value_type<T>()
+                                     && is_supported_representation_type<Rep>()>* = nullptr>
+    explicit fixed_point(T const& value, scale_type const& scale) :
+        _value{static_cast<Rep>(detail::shift_with_precise_round<Rep, Rad>(value, scale))},
+        _scale{scale}
+    {
+    }
+
+    /**---------------------------------------------------------------------------*
+    * @brief Constructor that will not perform shifting (assumes value already
+    * shifted)
+    *
+    * @param s scaled_integer that contains scale and already shifted value
+    *---------------------------------------------------------------------------**/
+    explicit fixed_point(scaled_integer<Rep> s) :
+        _value{s.value},
+        _scale{s.scale}
+    {
+    }
+
+   /**---------------------------------------------------------------------------*
+    * @brief Default onstructor that constructs `fixed_point` number with a
+    * value and scale of zero
+    *---------------------------------------------------------------------------**/
+    fixed_point() :
```

Comment:
Can this just be ?
```suggestion
    fixed_point() = default;
```

Reply:
No, because `strong_typedef` doesn't have a default ctor.